modules/core/web/src/components/ModelingTool.vue

```diff
@@ -1,31 +1,41 @@
 <template>
-  <div class="container" :class="{ 'not-clickable': !props.isEditable }">
+ <div class="container" :class="{ 'not-clickable': !props.isEditable }" @contextmenu="showContextMenu">
     <ModalAddAttributes />
     <div ref="modelingContainer" class="modelingContainer" @click.self="unselectAll">
       <EntityMain v-for="entity in entities.diagram.value.entities" :key="entity.id" class="entity" :entity="entity" :is-editable="props.isEditable" />
-
       <LineMain v-for="line in lineList" :key="line.id" :line="line" />
-
       <!-- Definiert global das aussehen der Pfeile -->
       <ArrowDefinitionVue class="svgMarker"></ArrowDefinitionVue>
+      <!-- Context menu -->
+      <div v-if="isContextMenuVisible" class="context-menu" :style="{ left: contextMenuPosition.x, top: contextMenuPosition.y }">
+        <div><IconEntity @mousedown="addElement($event, EntityType.ENTITY)" class="context-menu-item"  /></div>
+        <div><IconRelationshiptyp @mousedown="addElement($event, EntityType.RELATIONSHIP)"  class="context-menu-item" /></div>
+       <div> <IconEntityRelationshiptyp  @mousedown="addElement($event, EntityType.ENTITYRELATIONSHIP)" class="context-menu-item" /></div>
+       
+      </div>
     </div>
   </div>
 </template>
 
 <script setup lang="ts">
 import EntityMain from "./modelingTool/EntityMain.vue";
 import LineMain from "./modelingTool/LineMain.vue";
-
 import ArrowDefinitionVue from "./modelingTool/ArrowDefinition.vue";
 import ModalAddAttributes from "../dialog/DialogAddAttributes.vue";
-
-import { onMounted, reactive, computed, ref } from "vue";
+import { ref } from "vue";
+import { onMounted, reactive, computed } from "vue";
 import { useDiagramStore } from "../stores/diagramStore";
 import { useToolManagementStore } from "../stores/toolManagementStore";
-import { storeToRefs } from "pinia";
-
 import Connection from "../model/diagram/Connection";
 import Line from "../model/diagram/Line";
+import { defineProps } from "vue";
+import { storeToRefs } from "pinia";
+import IconEntity from "@/components/icons/IconEntitytyp.vue";
+import IconRelationshiptyp from "@/components/icons/IconRelationshiptyp.vue";
+import IconEntityRelationshiptyp from "@/components/icons/IconEntityRelationshiptyp.vue";
+import EntityType from "@/enums/EntityType";
+const isContextMenuVisible = ref(false);
+const contextMenuPosition = ref({ x: '0px', y: '0px' });
```

Comment:
Variables should never be defined under the imports. Please place them between the props and onMounted().